### ISSUE TITLE
Inconsistent syntax highlighting: destructuring let

### ISSUE THREAD

**Issue by lilred** [ionide/ionide-fsgrammar#24]
Repro:

create `test.fs`, and put the following code inside:

```
type T = C of int
let m = C 3
let (C n) = C 3
```

Expected behaviour: the `let` keywords should have the same color; `m` and `n` should have the same color.

Actual behavior: 
![image](https://cloud.githubusercontent.com/assets/1834919/22621959/c10190ea-eafd-11e6-90f1-81fac733e2b8.png)

`ionide-fsharp` 2.22.0, reproducible on OS X and Windows 10, VS Code stable and insiders.

**Comment by cloudRoutine** [comment]
When you find failing cases it's a big help just to PR them into [the sample code source files](https://github.com/ionide/ionide-fsgrammar/tree/master/sample-code) that way anyone working on the grammar can see everything that's broken and if they inadvertently break something.

**Comment by Krzysztof-Cieslak** [comment]
More examples:
```
let [| a; b |] = [| 1; 2 |]
let [ a; b ] = [1; 2]
let (a, b) = 1, 2
```
by @TeaDrivenDev in https://github.com/ionide/ionide-fsgrammar/issues/12#issuecomment-260971512

**Comment by Krzysztof-Cieslak** [comment]
Fixed by #30
